src/jit.rs

```diff
@@ -949,53 +973,71 @@ impl<'a> JitMemory<'a> {
+                                // This only affects the order of the move instructions, not the arguments.
+                                emit_rust_call(self, vtable.call, &[
+                                    Argument { index: 5, value: Value::Register(ARGUMENT_REGISTERS[4]) },
+                                    Argument { index: 4, value: Value::Register(ARGUMENT_REGISTERS[3]) },
+                                    Argument { index: 3, value: Value::Register(ARGUMENT_REGISTERS[2]) },
+                                    Argument { index: 2, value: Value::Register(ARGUMENT_REGISTERS[1]) },
+                                    Argument { index: 1, value: Value::Constant(unsafe { (*fat_ptr).data } as i64) }, // "&mut self" in the "call" method of the SyscallObject
+                                    Argument { index: 6, value: Value::Register(ARGUMENT_REGISTERS[5]) },
+                                    Argument { index: 7, value: Value::Register(R10) }, // memory_mapping
+                                ]);
+                            },
                         }
-                    }
 
-                    // Throw error if the result indicates one
-                    emit_jcc(self, 0x85, TARGET_PC_EPILOGUE);
-
-                    // Store Ok value in result register
-                    emit_load(self, OperandSize::S64, RDI, REGISTER_MAP[0], 8);
+                        // Throw error if the result indicates one
+                        emit_jcc(self, 0x85, TARGET_PC_EPILOGUE);
+
+                        // Store Ok value in result register
+                        emit_load(self, OperandSize::S64, RDI, REGISTER_MAP[0], 8);
+                    } else {
+                        match executable.lookup_bpf_call(insn.imm as u32) {
+                            Some(target_pc) => {
+                                emit_bpf_call(self, Value::Constant(*target_pc as i64));
+                            },
+                            None => executable.report_unresolved_symbol(insn_ptr)?,
+                        }
+                    }
+                },
+                ebpf::CALL_REG  => {
+                    unimplemented!();
+                    // emit_bpf_call(self, Value::Register(REGISTER_MAP[insn.imm as usize]));
                 },
-                ebpf::CALL_REG  => { unimplemented!() },
                 ebpf::EXIT      => {
-                    if insn_ptr != prog.len() / ebpf::INSN_SIZE - 1 {
-                        emit_jmp(self, TARGET_PC_EXIT);
-                    };
+                    emit_alu64_imm32(self, 0x81, 4, REGISTER_MAP[STACK_REG], !(CALL_FRAME_SIZE as i32 * 2 - 1)); // stack_ptr &= !(CALL_FRAME_SIZE * 2 - 1);
+                    emit_alu64_imm32(self, 0x81, 5, REGISTER_MAP[STACK_REG], CALL_FRAME_SIZE as i32 * 2); // stack_ptr -= CALL_FRAME_SIZE * 2;
+
+                    // if(stack_ptr < MM_STACK_START) goto exit;
```

Comment:
These types of comments that provide context are very helpful, especially for folks coming in fresh, more please :-)